app/lib/storacha.ts

```diff
@@ -0,0 +1,138 @@
+import { delegate } from '@ucanto/core'
+import { Principal } from '@ipld/dag-ucan'
+import * as Plan from '@storacha/capabilities/plan'
+import * as Usage from '@storacha/capabilities/usage'
+import * as Upload from '@storacha/capabilities/upload'
+import * as SSstore from '@storacha/capabilities/store'
+import * as Filecoin from '@storacha/capabilities/filecoin'
+import * as SpaceBlob from '@storacha/capabilities/space/blob'
+import * as SpaceIndex from '@storacha/capabilities/space/index'
+import { Client as StorachaClient } from '@storacha/client'
+import { MAX_FREE_BYTES } from './server/constants'
+import { AccountDID } from '@storacha/access'
+
+// default to 1 hour
+const defaultDuration = 1000 * 60 * 60
+const STORACHA_TRIAL_PLAN = 'did:web:trial.storacha.network'
+
+/**
+ * Fetches Storacha storage usage for a given space
+ * @param client - Storacha client instance
+ * @param space - Space DID to get usage for
+ * @returns Total storage usage in bytes
+ */
+export const getStorachaUsage = async (
+  client: StorachaClient,
+  space: `did:key:${string}`
+): Promise<number> => {
+  const now = new Date()
+  try {
+    const usage = await client.capability.usage.report(space, {
+      from: new Date(
+        now.getUTCFullYear(),
+        now.getUTCMonth() - 1,
+        now.getUTCDate(),
+        0,
+        0,
+        0,
+        0
+      ),
+      to: now,
+    })
+
+    // i don't think we allow people to create or own multiple
+    // spaces for now in the TG mini app
+    // if it happens in the future, we should account for it.
+    return Object.values(usage).reduce(
+      (sum, report) => sum + report.size.final,
+      0
+    )
+  } catch (err) {
+    console.error('Error while fetching usage report:', err)
+    throw new Error(`Failed to fetch usage report.`, { cause: err })
+  }
+}
+
+export const isPayingAccount = async (
+  client: StorachaClient,
+  accountDID?: AccountDID
+) => {
+  if (!client) throw new Error('Storacha client is not initialized')
+  try {
+    const account = Object.values(client.accounts())[0]
+
+    let planResult
+    if (account) {
+      planResult = await account.plan.get()
+    } else if (accountDID) {
```

Comment:
question: why wouldn't the account exist here? is there somewhere in the app where we need to check this but we're not logged in? I'm a little confused how this would work if the account wasn't logged in, wondering if this whole branch is unnecessary. 

suggestion: if it is required please add a comment for clarity

Reply:
This function is also used on the client side, in that case the user is logged in.

Reply:
Yes, the `if (account)` block is for the client side, and the `else if (accountDID)` is for the server side. I’ve updated the delegations the client makes to the server, adding a new one for `plan/get`, that’s why we now have the proof. This part is being handled manually

Reply:
ahhhh, right - we're using the delegation rather than an actual account login, got it. ok this is fine - it would be a bit better if the underlying library supported "setting up" an account from a delegation, and I suspect it might not be that hard to do, but we can ship this for now and make that improvement later. thanks!